doc/recipe/preprocessor.rst

```diff
@@ -180,12 +152,179 @@ steps:
+In some cases, preprocessor functions may work without supplementary variables,
+this is documented case by case in the preprocessor function definition.
+If a preprocessor function is used that uses supplementary variables, but they
+have not been specified in the recipe, they will be automatically added.
+If the automatic selection does not give the desired result, specify the
+supplementary variables in the recipe as described in
+:ref:`supplementary_variables`.
+
+By default, ancillary variables and cell measures will be removed from the
+variable before saving it to file because they can be as big as the main
+variable.
+To keep the ancillary variables and cell measures, disable the preprocessor
+function :func:`esmvalcore.preprocessor.remove_supplementary_variables` that
+removes them by setting ``remove_supplementary_variables: false`` in the
+preprocessor in the recipe.
+
+Examples
+--------
+
+Compute the global mean surface air temperature, while
+:ref:`automatically selecting the best matching supplementary dataset <supplementary_dataset_wildcards>`:
 
-Fx variables as cell measures or ancillary variables
-====================================================
-The following preprocessors may require the use of ``fx_variables`` to be able
-to perform the computations:
+.. code-block:: yaml
+
+  datasets:
+    - dataset: BCC-ESM1
+      project: CMIP6
+      ensemble: r1i1p1f1
+      grid: gn
+    - dataset: CanESM2
+      project: CMIP5
+      ensemble: r1i1p1,
+
+  preprocessors:
+    global_mean:
+      area_statistics:
+        operator: mean
+
+  diagnostics:
+    example_diagnostic:
+      description: Global mean temperature.
+      variables:
+        tas:
+          mip: Amon
+          preprocessor: global_mean
+          exp: historical
+          timerange: '1990/2000'
+          supplementary_variables:
+            - short_name: areacella
+              mip: fx
+              exp: '*'
+              activity: '*'
+              ensemble: '*'
+      scripts: null
+
+Attach the land area fraction as an ancillary variable to surface air
+temperature and store both in the same file:
+
+.. code-block:: yaml
+
+  datasets:
+    - dataset: BCC-ESM1
+      ensemble: r1i1p1f1
+      grid: gn
+
+  preprocessors:
+    keep_land_area_fraction:
+      remove_supplementary_variables: false
+
+  diagnostics:
+    example_diagnostic:
+      description: Attach land area fraction.
+      variables:
+        tas:
+          mip: Amon
+          project: CMIP6
+          preprocessor: keep_land_area_fraction
+          exp: historical
+          timerange: '1990/2000'
+          supplementary_variables:
+            - short_name: sftlf
+              mip: fx
+              exp: 1pctCO2
+      scripts: null
+
+
+Automatically define the required ancillary variable (``sftlf`` in this case)
+and cell measure (``areacella``), but do not use ``areacella`` for dataset
+``BCC-ESM1``:
+
+.. code-block:: yaml
```

Comment:
This third example does not work when running it with ESGF only and without using any local data. Lots of `areacella` files are found on ESGF but the ESGF request is somehow not working:
```bash
DEBUG   requests_cache.session:173 Pre-cache checks for response from https://esg-dn1.nsc.liu.se/esg-search/search?format=application%2Fsolr%2Bjson&limit=500&distrib=t\
rue&offset=0&type=File&project=CMIP5&model=CanESM2&ensemble=r0i0p0&experiment=historical&time_frequency=fx&institute=CCCma&cmor_table=fx&product=output&variable=areacella: {'disabled cache': False, 'disabl\
ed method': False, 'disabled status': False, 'disabled by filter': False, 'disabled by headers or expiration params': False}
2023-02-22 16:52:06,174 UTC [2163320] DEBUG   esmvalcore.esgf._search:155 Found the following files matching facets {'project': 'CMIP5', 'model': 'CanESM2', 'ensemble': 'r0i0p0', 'experiment': 'historical'\
, 'time_frequency': 'fx', 'institute': 'CCCma', 'cmor_table': 'fx', 'product': 'output', 'variable': 'areacella'}: none
2023-02-22 16:52:06,174 UTC [2163320] ERROR   esmvalcore._recipe.check:118 No input files found for Dataset:
{'dataset': 'CanESM2',
 'project': 'CMIP5',
 'mip': 'fx',
 'short_name': 'sftlf',
 'ensemble': 'r0i0p0',
 'exp': 'historical',
 'frequency': 'fx',
 'institute': 'CCCma',
 'long_name': 'Land Area Fraction',
 'modeling_realm': ['atmos'],
 'original_short_name': 'sftlf',
 'product': 'output',
 'standard_name': 'land_area_fraction',
 'units': '%'}
```

Reply:
It the thing with the CMIP5 CanESM2 institute, the facet value is `CCCMA` on ESGF instead of `CCCma`. I will update the example and ask the folks at ESGF if there is any way to correct this.

Reply:
That works fine now, thanks!